appintro/src/main/java/com/github/paolorotolo/appintro/AppIntroViewPager.kt

```diff
@@ -97,45 +96,68 @@ class AppIntroViewPager(context: Context, attrs: AttributeSet) : ViewPager(conte
+    {
+        if (!handleTouchEvent(event)) {
+            return false
+        }
+
+        // Calling super will allow the slider to "work" left and right.
+        return super.onInterceptTouchEvent(event)
+    }
 
     override fun onTouchEvent(event: MotionEvent?): Boolean {
+        if (!handleTouchEvent(event)) {
+            return false
+        }
+
+        // Calling super will allow the slider to "work" left and right.
+        return super.onTouchEvent(event)
+    }
+
+    private fun handleTouchEvent(event: MotionEvent?) : Boolean {
         // If paging is disabled we should ignore any viewpager touch
         // (also, not display any error message)
         if (!isFullPagingEnabled) {
             return false
         }
 
-        val canRequestNextPage = onNextPageRequestedListener?.onCanRequestNextPage() ?: true
-        when (event?.action) {
-            MotionEvent.ACTION_UP -> performClick()
-            MotionEvent.ACTION_DOWN -> currentTouchDownX = event.x
-            MotionEvent.ACTION_MOVE -> {
+        when(event?.action) {
+            MotionEvent.ACTION_DOWN -> {
+                currentTouchDownX = event.x
+                currentTouchDownY = event.y
+            }
+            else -> {
+                val canRequestNextPage = onNextPageRequestedListener?.onCanRequestNextPage() ?: true
+
                 // If user can't request the page, we shortcircuit the ACTION_MOVE logic here.
                 // We need to return false, and also call onIllegallyRequestedNextPage if the
                 // threshold was too high (so the user can be informed).
                 if (!canRequestNextPage) {
-                    if (userIllegallyRequestNextPage(event)) {
-                        onNextPageRequestedListener?.onIllegallyRequestedNextPage()
+
+                    event?.let {
+                        if(userIllegallyRequestNextPage(event)) {
+                            onNextPageRequestedListener?.onIllegallyRequestedNextPage()
+                        }
                     }
+
                     return false
                 }
             }
         }
 
-        // Calling super will allow the slider to "work" left and right.
-        return super.onTouchEvent(event)
+        return true
     }
 
     /**
      * Util function to check if the user Illegaly request a swipe.
      * Also checks if the event happened not earlier than every 1000ms
      */
     private fun userIllegallyRequestNextPage(event: MotionEvent): Boolean {
-        if (Math.abs(event.x - currentTouchDownX) >= VALID_SWIPE_THRESHOLD_PX) {
-            if (System.currentTimeMillis() - illegallyRequestedNextPageLastCalled >=
-                ON_ILLEGALLY_REQUESTED_NEXT_PAGE_MAX_INTERVAL
-            ) {
+        if ((event.x - currentTouchDownX).absoluteValue >= VALID_SWIPE_THRESHOLD_PX_X
+                && (event.y - currentTouchDownY).absoluteValue <= VALID_SWIPE_THRESHOLD_PX_Y) {
```

Comment:
Why exactly do we also need Y-Coordinate? Viewpager only supports Horizontal navigation.

Reply:
@AnuthaDev It prevents detecting a false positive sliding attempt. Think of a scroll view within a slide. If one wants to scroll the scroll view vertically but also scrolls a bit horizontally (by accident), a sliding attempt would be assumed (if the user scrolled more than 25px horizontally during the intended vertical scroll).